plugins/modules/proxmox_storage.py

```diff
@@ -0,0 +1,520 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2025, Florian Paul Azim Hoberg (@gyptazy) <florian.hoberg@credativ.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: proxmox_storage
+version_added: 1.2.0
+short_description: Manage storage in PVE clusters and nodes
+description:
+  - Manage storage in PVE clusters and nodes.
+author: Florian Paul Azim Hoberg (@gyptazy)
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  node_name:
+    description:
+      - The targeted Proxmox VE node to perform actions on.
+    type: str
+    required: true
+  nodes:
+    description:
+      - A list of Proxmox VE nodes where the storage action should be performed on.
+    type: list
+    elements: str
+    required: true
+  name:
+    description:
+      - The name of the storage displayed in the storage list.
+    type: str
+    required: true
+  state:
+    description:
+      - The state of the defined storage type to perform.
+    choices: ["present", "absent"]
+    type: str
+  type:
+    description:
+      - The storage type/protocol to use when adding the storage.
+    type: str
+    required: true
+    choices: ['cephfs', 'cifs', 'iscsi', 'nfs', 'pbs']
+  cephfs_options:
+    description:
+      - Extended information for adding CephFS storage.
+    type: dict
+    suboptions:
+      monhost:
+        description:
+          - The hostname or IP address of the monhost.
+        type: str
+        required: false
+      username:
+        description:
+          - The required username for the storage system.
+        type: str
+        required: false
+      password:
+        description:
+          - The required password for the storage system.
+        type: str
+        required: false
+      path:
+        description:
+          - The path to be used within the CephFS.
+        type: str
+        default: '/'
+        required: false
+      subdir:
+        description:
+          - The subdir to be used within the CephFS.
+        type: str
+        required: false
+      client_keyring:
+        description:
+          - The client keyring to be used.
+        type: str
+        required: false
```

Comment:
However, it's good to know that this will not create a new Ceph pool to store the data in. It will rather use an existing pool (the `fs-name` parameter in the API), which is missing

```suggestion
        required: false
      fs_name:
        description:
          - The Ceph filesystem name
        type: str
        required: false
```